From a537107cc8903895cd5118ef2bf95fb6f52f5856 Mon Sep 17 00:00:00 2001
From: Yifeng Zhao <zyf@rock-chips.com>
Date: Thu, 23 Aug 2018 16:34:08 +0800
Subject: [PATCH] drivers: rk_nand: fix android 9.0 block device count not
 mount issue

The android 9.0 scan block device type: platform, vbd and pci.
The old nand driver create nand device uevent in "sys/devices/
virtual/", android 9.0 could not support.
update new driver create nand device uevent in "sys/devices/
platform/ff3b0000.nandc/block/rknand0" to fix the issuex.

Change-Id: Idda634cc5584fb4a8b531fbbea481d1a08c63cca
Signed-off-by: Yifeng Zhao <zyf@rock-chips.com>
---
 drivers/rk_nand/rk_nand_blk.c | 1 +
 drivers/rk_nand/rk_nand_blk.h | 1 +
 2 files changed, 2 insertions(+)

diff --git a/drivers/rk_nand/rk_nand_blk.c b/drivers/rk_nand/rk_nand_blk.c
index d015b511f0d2..967ec4ed2562 100644
--- a/drivers/rk_nand/rk_nand_blk.c
+++ b/drivers/rk_nand/rk_nand_blk.c
@@ -586,6 +586,7 @@ static int nand_add_dev(struct nand_blk_ops *nandr, struct nand_part *part)
 			 part->name);
 	} else {
 		gd->flags = GENHD_FL_EXT_DEVT;
+		gd->driverfs_dev = g_nand_device;
 		gd->minors = 255;
 		snprintf(gd->disk_name,
 			 sizeof(gd->disk_name),
diff --git a/drivers/rk_nand/rk_nand_blk.h b/drivers/rk_nand/rk_nand_blk.h
index 72b3f505abf8..dc4776fc5d6f 100644
--- a/drivers/rk_nand/rk_nand_blk.h
+++ b/drivers/rk_nand/rk_nand_blk.h
@@ -48,6 +48,7 @@ struct nand_blk_ops {
 	struct module *owner;
 };
 
+extern struct device *g_nand_device;
 void rknand_dev_suspend(void);
 void rknand_dev_resume(void);
 void rknand_dev_shutdown(void);
-- 
2.35.3

